lang/LangPrimSource/PyrFilePrim.cpp

```diff
@@ -1326,6 +1326,74 @@ int prPipeOpen(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int prPipeOpenArgv(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a, *b, *c;
+	char mode[12];
+
+	a = g->sp - 2;
+	b = g->sp - 1; //argv
+	c = g->sp;     //mode
+
+	if (NotObj(b)) return errWrongType;
+
+	if (NotObj(c) || !isKindOf(slotRawObject(c), class_string))
+		return errWrongType;
+	if (slotRawObject(c)->size > 11) return errFailed;
+
+	PyrFile *pfile = (PyrFile*)slotRawObject(a);
+
+	PyrObject *obj = slotRawObject(b);
+	if (!(slotRawInt(&obj->classptr->classFlags) & classHasIndexableInstances))
+		return errNotAnIndexableObject;
+
+	if( obj->size < 1)
+		return errFailed;
+
+	memcpy(mode, slotRawString(c)->s, slotRawObject(c)->size);
+	mode[slotRawString(c)->size] = 0;
+
+	PyrSlot filenameSlot;
+	getIndexedSlot(obj, &filenameSlot, 0);
+	if (!isKindOfSlot(&filenameSlot, class_string)) return errWrongType;
+	char filename[PATH_MAX];
+	if (slotRawObject(&filenameSlot)->size > PATH_MAX - 1) return errFailed;
+	slotStrVal(&filenameSlot, filename, slotRawObject(&filenameSlot)->size + 1);
+
+	std::vector<char *> argv (obj->size + 1);
+
+	bfs::path p;
+	p /= filename;
+	std::string filenameOnly = p.filename().string();
+	std::vector<char> vfilenameOnly(filenameOnly.begin(), filenameOnly.end());
+	vfilenameOnly.push_back('\0');
+
+	argv[0] = vfilenameOnly.data();
+	argv[obj->size] = NULL;
+
+	if(obj->size > 1) {
```

Comment:
this check is unnecessary - line 1350 early-exits if this is false, and the for-loop below wouldn't run if `obj->size <= 1`

Reply:
ok